.github/workflows/docker_image.yml

```diff
@@ -0,0 +1,58 @@
+name: Docker Image
+
+on:
+  push:
+    branches: [ 'devnet_*', 'testnet_*' ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest-16-cores
+    timeout-minutes: 60
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Auth service account
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: ${{ secrets.GCP_SA_ACTIONS_RUNNER_KEY }}
+      - name: Set up Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@v1
+    
+      - name: Authenticate Docker with GCP
+        run: |
+          gcloud auth configure-docker us-docker.pkg.dev
+  
+      - name: Set env variables
+        run: |
+          echo "GIT_COMMIT=${GITHUB_SHA}" >> $GITHUB_ENV
+          if [ -n "${{ github.head_ref }}" ]; then
+            echo "BRANCH_NAME=${{ github.head_ref }}" >> $GITHUB_ENV
+          else
+            echo "BRANCH_NAME=${{ github.ref_name }}" >> $GITHUB_ENV
+          fi
+
+      - name: Set docker image env variable
+        run: |
+          echo "DOCKER_IMAGE=us-docker.pkg.dev/linera-io-dev/linera-public-registry/linera-${{ env.BRANCH_NAME }}" >> $GITHUB_ENV
```

Comment:
Thought it made sense to encode the branch name here in the image name, so we can use latest to easily get the latest image from a branch

Reply:
Shouldn't we keep some limited history and have the (short) commit hash in the name?

Perhaps every commit is too much by the way. Could it be every hour or so?

Reply:
I think I can limit how many images are stored in the GCP configs, I can look into that. I was thinking of keeping only the last X images, or something like that.
We're tagging the image with the commit hash, so it's easier for us to find for example the latest image for `linera-testnet-boole`, and know what commit hash it was built with because we have the tag. I can change it to the short tag though! That's what I had originally.

The problem with not doing every commit is that if we want to release, and that hour's build still didn't run, we won't have the image ready, and will fall into the same issue we had before, having to build the image from scratch. I think limiting how many we're storing in GCP should be enough 🤔
Keep in mind this is only for commits to the `testnet_*` and `devnet_*` branches, so it shouldn't be a huge amount.

Reply:
PR with cleanup policy: https://github.com/linera-io/linera-infra/pull/98